File: modules/expo-android-alarm-manager/src/ExpoAndroidAlarmManagerModule.ts
import { NativeModule, requireNativeModule } from "expo";

import { ExpoAndroidAlarmManagerModuleConstants } from "./ExpoAndroidAlarmManager.types";

declare class ExpoAndroidAlarmManagerModule extends NativeModule {
  SHARED_PREFS_NAME: string;
  KEY_APP_LIST: string;
  registerAlarm(hour: number, minute: number): Promise<void>;
  cancelAlarm(): Promise<void>;
  testAlarm(): Promise<void>;
  getAppList(): Promise<string[]>;
  setAppList(apps: string[]): Promise<void>;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<ExpoAndroidAlarmManagerModule>(
  "ExpoAndroidAlarmManager",
);

File: modules/expo-android-alarm-manager/ios/ExpoAndroidAlarmManagerModule.swift
import ExpoModulesCore

public class ExpoAndroidAlarmManagerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAndroidAlarmManager")

    AsyncFunction("registerAlarm") { (hour: Int, minute: Int) in
      throw Exception(name: "UnsupportedPlatform", description: "Alarm manager is only supported on Android")
    }

    AsyncFunction("cancelAlarm") {
      throw Exception(name: "UnsupportedPlatform", description: "Alarm manager is only supported on Android")
    }

    AsyncFunction("testAlarm") {
      throw Exception(name: "UnsupportedPlatform", description: "Alarm manager is only supported on Android")
    }

    AsyncFunction("getAppList") { () -> [String] in
      throw Exception(name: "UnsupportedPlatform", description: "Alarm manager is only supported on Android")
    }

    AsyncFunction("setAppList") { (apps: [String]) in
      throw Exception(name: "UnsupportedPlatform", description: "Alarm manager is only supported on Android")
    }
  }
}
